src/devices_models/devices/hydro_generation.jl

```diff
@@ -315,101 +315,55 @@ function cost_function(psi_container::PSIContainer,
-        device_budget_param_ub(psi_container,
-                            budget_data,
-                            Symbol("budget_$(H)"), # TODO: better name for this constraint
-                            UpdateRef{H}(:storagecapacity),
-                            Symbol("P_$(H)"))
-    else
-        device_budget_param_ub(psi_container,
-                            budget_data,
-                            Symbol("budget_$(H)"), # TODO: better name for this constraint
-                            Symbol("P_$(H)"))
-    end
+
+    return
 end
 
-function device_budget_param_ub(psi_container::PSIContainer,
-                            budget_data::Vector{Tuple{String, Int64, Float64, Vector{Float64}}},
-                            cons_name::Symbol,
-                            param_reference::UpdateRef,
-                            var_name::Symbol)
-    time_steps = model_time_steps(psi_container)
-    variable = get_variable(psi_container, var_name)
-    set_name = (r[1] for r in budget_data)
-    no_of_budgets = length(budget_data[1][4])
-    time_lengths = time_steps/length(budget_data[1][4])
-    time_chunks = reshape(time_steps, (time_lengths, no_of_budgets))
-    constraint = _add_cons_container!(psi_container, cons_name, set_name, no_of_budgets)
-    param = _add_param_container!(psi_container, param_reference, names, no_of_budgets)
-
-    for data in budget_data, i in 1:no_of_budgets
-        name = data[1]
-        forecast = data[4][i]
-        multiplier = data[3]
-        param[name] = PJ.add_parameter(psi_container.JuMPmodel, forecast)
-        constraint[name] = JuMP.@constraint(psi_container.JuMPmodel,
-                    sum([variable[name, t] for t in time_chunks[:, i]]) <= multiplier*param[name])
+function energylimit_constraints!(psi_container::PSIContainer,
+                    devices::IS.FlattenIteratorWrapper{H},
+                    model::DeviceModel{H, <:AbstractHydroFormulation},
+                    system_formulation::Type{<:PM.AbstractPowerModel},
+                    feed_forward::EnergyLimitFF) where H<:PSY.HydroGen
+
+    parameters = model_has_parameters(psi_container)
+    for prefix in get_affected_variables(feed_forward)
+        var_name = Symbol(prefix, "_$(H)")
+        param_reference = UpdateRef{JuMP.VariableRef}(var_name)
+        
+        if parameters
+            device_energylimit(psi_container,
+                                param_reference,
+                                Symbol("energy_limit_$(H)"), # TODO: better name for this constraint
+                                Symbol("P_$(H)"))
+        else
+            @warn("Simulation without parameters doesn't support EnergyLimitFF, please consider changing your simulation setup.")
+        end
     end
 
     return
 end
 
-
-function device_budget_ub(psi_container::PSIContainer,
-                            budget_data::Vector{Tuple{String, Int64, Float64, Vector{Float64}}},
+function device_energylimit(psi_container::PSIContainer,
```

Comment:
Same here, why are you passing a parameter ref here? you need to detect if the model uses parameters or not like all the other model implementations and then implement the constraint appropriately